### ISSUE TITLE
Solution Package Export Error Caused by Security Role Names with Special Characters

### ISSUE THREAD

**Issue by cbritton123** [jlattimer/CRMDeveloperExtensions#83]
Hi again,
An error is generated by the CRM Developer Extensions when exporting a solution package if the Security Role name contains special characters such as colon (':') or slash ('/'). Dynamics CRM does allow System Administrator users to define security role names containing these special characters and these roles are exported correctly in the solution package's customizations.xml file in the solution package ZIP file. The problem lies within the CRM Developer Extensions tool itself and the result is that the security role's corresponding XML file (extracted from the customizations.xml file) is not generated under the Roles folder. See log extract below:

> 2016-11-18 08:01:44.951 - Info    - SPAC: administrateur-administratrice des utilisateurs (PSPC: User Administrator)
2016-11-18 08:01:44.982 - Error  The given path's format is not supported.
2016-11-18 08:01:44.982 - Error  See log file 'C:\Users\parentr3\Documents\Visual Studio 2012\Projects\PspcSolution(1)\GlobalSolution\SolutionPackager.log' for details.
2016-11-18 08:01:45.182 - Error  System.NotSupportedException: The given path's format is not supported.
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost)
   at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost)
   at System.IO.File.InternalWriteAllText(String path, String contents, Encoding encoding, Boolean checkHost)
   at System.IO.File.WriteAllText(String path, String contents, Encoding encoding)
   at Microsoft.Crm.Tools.SolutionPackager.Helper.WriteToFile(String filename, String content, Encoding encoding)
   at Microsoft.Crm.Tools.SolutionPackager.Helper.WriteToFile(String filename, XElement element)
   at Microsoft.Crm.Tools.SolutionPackager.ComponentProcessorBase.WriteToFiles(ComponentCollection components)
   at Microsoft.Crm.Tools.SolutionPackager.DiskWriter.WriteComponents()
   at Microsoft.Crm.Tools.SolutionPackager.SolutionPackager.Run(IPackageReader reader, IPackageWriter writer)

**Comment by jlattimer** [comment]
I have a suspicion this might actually be an issue with the version of the solution packager you're using as it handles extracting and naming the files. The code I've wrapped around it only processes the files after they're extracted. I did a test with a role using a ":" in the name and used SolutionPackager.exe from the 8.2 SDK and it extracted fine.

![image](https://cloud.githubusercontent.com/assets/1769982/20460511/ebb65288-aeaa-11e6-8afe-8a3f8ec77bd3.png)
